### PR TITLE
 44-in-vrdr_causeofdeath_coding_update-message-the-messageheader-does-not-have-a-"source"-data-element

### PR DESCRIPTION
**Pull request by consultantusa**

### CODE REVIEW

**Comment by consultantusa** [comment]
Bugs were due mostly to wrong order of params in constructors. They were fixed.

**Comment by consultantusa** [comment]
I am working on the issue. Thank you.

**Comment by consultantusa** [comment]
I have checked the example files for these message types, and they were tested in MessagingTest.java, just that in order to get the messageSource, DeathRecordSubmissionMessage must be used as the wrapper first, as shown in my additions to the tests

**Comment by johnarleyburns** [comment]
Okay so it ready for review?

File: src/main/java/edu/gatech/chai/VRDR/messaging/CauseOfDeathCodingMessage.java [hunk 1]
```diff
@@ -44,8 +44,8 @@ public void setCauseOfDeathCodedContentBundle(CauseOfDeathCodedContentBundle cau
 
     public CauseOfDeathCodingMessage(BaseMessage messageToCode) {
         this(messageToCode == null ? null : messageToCode.getMessageHeaderId(),
-                messageToCode == null ? null : messageToCode.getMessageDestination(),
```
Comment (johnarleyburns) [inline src/main/java/edu/gatech/chai/VRDR/messaging/CauseOfDeathCodingMessage.java:47]: This order should be kept and not flipped.  It confused me at first, but the coding messages are reverse flow back to the jurisdiction.

**Review by johnarleyburns — CHANGES_REQUESTED** [review]

File: src/main/java/edu/gatech/chai/VRDR/messaging/DemographicsCodingUpdateMessage.java [hunk 2]
```diff
@@ -40,8 +40,8 @@ public void setDemographicCodedContentBundle(DemographicCodedContentBundle demog
 
     public DemographicsCodingUpdateMessage(BaseMessage messageToCode) {
         this(messageToCode == null ? null : messageToCode.getMessageHeaderId(),
-                messageToCode == null ? null : messageToCode.getMessageDestination(),
-                messageToCode == null ? null : messageToCode.getMessageSource());
+		messageToCode == null ? null : messageToCode.getMessageSource(),
```
Comment (johnarleyburns) [inline src/main/java/edu/gatech/chai/VRDR/messaging/DemographicsCodingUpdateMessage.java:43]: Don’t flip this one it’s meant to be this way

**Review by johnarleyburns — CHANGES_REQUESTED** [review]
This one may actually be an NCHS API issue and not a library issue, please conduct more investigation with NCHS

File: src/main/java/edu/gatech/chai/VRDR/messaging/DemographicsCodingUpdateMessage.java (hunk 2 above)
Comment (consultantusa) [reply src/main/java/edu/gatech/chai/VRDR/messaging/DemographicsCodingUpdateMessage.java:43]: I reverted the changes, and found out that the example files in question all lack the "source/endpoint" attribute.

**Review by johnarleyburns — CHANGES_REQUESTED** [review]
Lets go back to the originator of the issue, and get more information.

**Comment by johnarleyburns** [comment]
Issue resolved on server side, no code change needed.

**Pull request closed without merging**